packages/server/src/queue/queue-sse.service.ts

```diff
@@ -43,15 +43,15 @@ export class QueueSSEService {
   updateQueue = this.throttleUpdate(async (queueId) => {
     const queue = await this.queueService.getQueue(queueId);
     if (queue) {
-      this.sendToRoom(queueId, () => ({ queue }));
+      await this.sendToRoom(queueId, async () => ({ queue }));
     }
   });
 
-  private sendToRoom(
+  private async sendToRoom(
     queueId: number,
-    data: (metadata: QueueClientMetadata) => SSEQueueResponse,
+    data: (metadata: QueueClientMetadata) => Promise<SSEQueueResponse>,
   ) {
-    this.sseService.sendEvent(idToRoom(queueId), data);
+    await this.sseService.sendEvent(idToRoom(queueId), data);
```

Comment:
Just for clarification, why do we need to `await` this?

Reply:
@dajinchu

Reply:
sendEvent is async now, because the `data` callback is async. All this happened because `personalizeQuestions` has a database query inside it that makes it async https://github.com/sandboxnu/office-hours/blob/bdd228776ae4d11dbe7b3333f41a4e9c6bec2690/packages/server/src/queue/queue.service.ts#L85